e2e/google_analytics.spec.js

```diff
@@ -0,0 +1,32 @@
+import { test, expect } from "@playwright/test";
+import { VISUAL_EDITOR_URL } from "./helpers/constants.js";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(VISUAL_EDITOR_URL);
+});
+
+test("menu select emit tracking events", async ({ page }) => {
+  const selectTextPromise = page.evaluate(() => {
+    return new Promise((resolve) => {
+      document.addEventListener("visualEditorSelectChange", (e) =>
+        resolve(e.detail.selectText),
+      );
+    });
+  });
+  page.locator('select:has-text("Paragraph")').selectOption("Heading 2");
+  const selectText = await selectTextPromise;
+  await expect(selectText).toEqual("Heading 2");
+});
+
+test("menu buttons emit tracking events", async ({ page }) => {
+  const buttonTextPromise = page.evaluate(() => {
+    return new Promise((resolve) => {
+      document.addEventListener("visualEditorButtonClick", (e) =>
+        resolve(e.detail.buttonText),
+      );
+    });
+  });
+  await page.getByTitle("Bullet list").click();
+  const buttonText = await await await buttonTextPromise;
```

Comment:
This... doesn't look right

Reply:
Aha, can you sense I was struggling with the tests a bit 😂